Framework/Parallel/test/EventParserTest.h

```diff
@@ -0,0 +1,464 @@
+#ifndef MANTID_PARALLEL_COLLECTIVESTEST_H_
+#define MANTID_PARALLEL_COLLECTIVESTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidParallel/IO/EventParser.h"
+
+#include <numeric>
+
+using namespace Mantid;
+using namespace Parallel::IO;
+using Mantid::Types::Core::DateAndTime;
+using Mantid::Types::Event::TofEvent;
+
+namespace detail {
+template <typename IndexType, typename TimeZeroType, typename TimeOffsetType>
+class FakeParserDataGenerator {
```

Comment:
Naming is good. But would be nice to have brief comments somewhere on how this is generating the events for use by the parser. Looks like there is quite a bit of logic in this type.